test/test_unix_socket/receiver.py

```diff
@@ -23,23 +23,18 @@ def run_dummy_listener(socket_path=DEFAULT_UNIX_SOCKET_PATH):
         client_socket.connect(socket_path)
         print("Connected to server.")
 
+        buffer = ""
         while True:
-            # Use select to wait until there is data to read on the socket
-            readable, _, _ = select.select([client_socket], [], [])
-
-            for sock in readable:
-                # Receive data from the server
-                received_data = sock.recv(1024)
-
-                # If no data is received, close the connection
-                if not received_data:
-                    print("No data received. Closing connection.")
-                    client_socket.close()
-                    exit()
-
-                # Decode the received data from bytes to JSON
-                json_data = json.loads(received_data.decode())
+            # Receive data from the server
+            buffer = buffer + client_socket.recv(1024).decode()
```

Comment:
I would explicitly specify the encoding

Reply:
Well, yeah. Default is utf-8. I will watch out if somebody might start using Windows-1256 ;)